imageio/imageio-webp/src/main/java/com/twelvemonkeys/imageio/plugins/webp/lossless/transform/PredictorTransform.java

```diff
@@ -0,0 +1,238 @@
+package com.twelvemonkeys.imageio.plugins.webp.lossless.transform;
+
+import java.awt.image.Raster;
+import java.awt.image.WritableRaster;
+
+import static java.lang.Math.*;
+
+public class PredictorTransform implements Transform {
+    private final Raster data;
+    private final byte bits;
+
+    public PredictorTransform(Raster raster, byte bits){
+        this.data = raster;
+        this.bits = bits;
+    }
+
+    private static byte[] select(final byte[] l, final byte[] t, final byte[] tl){
+        // l = left pixel, t = top pixel, tl = top left pixel.
+
+        // ARGB component estimates for prediction.
+
+        int pAlpha = addSubtractFull(l[3], t[3], tl[3]);
+        int pRed = addSubtractFull(l[0], t[0], tl[0]);
+        int pGreen = addSubtractFull(l[1], t[1], tl[1]);
+        int pBlue = addSubtractFull(l[2], t[2], tl[2]);
+
+        // Manhattan distances to estimates for left and top pixels.
+        int pL = mannhattanDistance(l, pAlpha, pRed, pGreen, pBlue);
+        int pT = mannhattanDistance(t, pAlpha, pRed, pGreen, pBlue);
+
+        // Return either left or top, the one closer to the prediction.
+        return pL < pT ? l : t;
+    }
+
+    private static int mannhattanDistance(byte[] rgba, int pAlpha, int pRed, int pGreen, int pBlue){
```

Comment:
Small typo in Manhattan. 😉 

```suggestion
    private static int manhattanDistance (byte[] rgba, int pAlpha, int pRed, int pGreen, int pBlue) {
```